ruby/object_oriented_programming_basics/object_oriented_programming.md

```diff
@@ -2,42 +2,34 @@
-  3. Read through [Launch School's OOP book](https://launchschool.com/books/oo_ruby/read/introduction) for a more thorough understanding.
-  4. Read through these reinforcing posts by Erik Trautman to help you answer the questions in the "Learning Outcomes" section:
-      1. [Ruby Explained: Classes](http://www.eriktrautman.com/posts/ruby-explained-classes)
-      2. [Ruby Explained: Inheritance and Scope](http://www.eriktrautman.com/posts/ruby-explained-inheritance-and-scope)
-  5. Read the article [Object Relationships in Basic Ruby](https://medium.com/@marcellamaki/object-relationships-in-basic-ruby-1af5773fff48) to see an example of how two classes can interact.
-  6. Read the [Bastard's Chapter on Error Handling](http://ruby.bastardsbook.com/chapters/exception-handling/) to reinforce your understanding of dealing with errors.
-  7. Do [Quiz #5](http://www.codequizzes.com/ruby/beginner/intro-object-oriented-programming) from [Code Quizzes](http://www.codequizzes.com).
-  8. Do [Quiz #7](http://www.codequizzes.com/ruby/beginner/modules-classes-inheritance) as well.
-  9. Glance over the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide) so you have an idea of how to make your code look more professional. It is recommended to install [rubocop](https://docs.rubocop.org/rubocop/installation.html), a static code analyzer (linter) and formatter, which is based on this style guide. 
-      1. Read the [basic usage](https://docs.rubocop.org/rubocop/usage/basic_usage.html) of rubocop in the terminal. 
-      2. To highlight the rubocop offenses in VSCode, you will need to have the 'Ruby' extension installed. In addition, you will need to update your settings.json file with the following lines:
-
+1. Read the [Object Oriented Programming with Ruby](https://launchschool.com/books/oo_ruby) online book, by Launch School.
+1. Read through these reinforcing posts by Erik Trautman to help you answer the questions in the "Learning Outcomes" section:
+    * [Ruby Explained: Classes](http://www.eriktrautman.com/posts/ruby-explained-classes)
+    * [Ruby Explained: Inheritance and Scope](http://www.eriktrautman.com/posts/ruby-explained-inheritance-and-scope)
+1. Read the article [Object Relationships in Basic Ruby](https://medium.com/@marcellamaki/object-relationships-in-basic-ruby-1af5773fff48) to see an example of how two classes can interact.
+1. Read the [Bastard's Chapter on Error Handling](http://ruby.bastardsbook.com/chapters/exception-handling/) to reinforce your understanding of dealing with errors.
+1. Do [Quiz #5](http://www.codequizzes.com/ruby/beginner/intro-object-oriented-programming) and [Quiz #7](http://www.codequizzes.com/ruby/beginner/modules-classes-inheritance) from Code Quizzes
+1. Every programming language community develops a style guide to help make code more maintainable and easier to read, therefore it is important to familiarize yourself with the [Ruby Style Guide](https://rubystyle.guide/).
```

Comment:
I know you've just moved this, but what do you think about taking linting out of this lesson?

I feel like we're conflating OOP and linters here. Theres already a lot to take in with OOP, and while we only have this one "god" lesson for OOP atm, it would be a much better learning experience to focus on just OOP here.

Is there somewhere further into the Ruby course we could introduce linters instead? I know theres an issue for a full on [Ruby linting lesson](https://github.com/TheOdinProject/curriculum/issues/23802).

Whats your thoughts?

Reply:
It would be much better in it's own lesson, but I think it is essential to start using it for Tic Tac Toe. I could go with moving it to the Tic Tac Toe assignment, or just leaving it put (until we have the linting lesson).

We could move it to the beginning of the Ruby basics exercises, but would require a bit more work as at that point people could still be just using irb or replit for their code. So that section would need to be fleshed out a bit more for linting to make sense.

Reply:
Just so I can understand your perspective better, why is it essential for the Tic Tac Toe project?

Reply:
One of the reasons that I think it is essential is because it reminds people not write methods that are too long or has too many assignment branches. Before we started recommending rubocop, people often had these huge methods that were doing a lot of different things. Which I see as basically functional programming in a class wrapper.

Rubocop helps remind people in "real time", which is think is important as they go into this first project.

Reply:
Thanks for explaining, that does make a lot of sense. I think a lot of those problems might be a consequence how we're teaching OOP. But happy to revisit that when we eventually get around to rewriting this section lol.

:shipit:

Reply:
Yes, totally! We are on the same page, wanting our lessons to touch base of these fundamental practices.